### ISSUE TITLE
MiFlora plant sensor - No data

### ISSUE THREAD

**Issue by Krojack** [custom-components/ble_monitor#290]
I just got a plant/soil sensor. It's model HHCCJCY01 and that's what's showing up in HA for the 4 entities that get created. I just don't get any data populated from the sensor. 

The sensor is branded as "[Northfifteen Connected Home Plant Monitor](https://www.amazon.com/gp/product/B07L491F8K/)". I installed their app and the sensor is working though their interface. The ONLY thing I haven't done is update the firmware. It's currently running v2.7.0 but I have the option to update to v3.2.2.  

![Configuration-Home-Assistant](https://user-images.githubusercontent.com/908461/108915939-93180c80-75fb-11eb-9b14-c77dffc0ecca.jpg)

Config entries:
```
devices:
  - mac: 'C4:7C:8D:63:C9:E8'
    name: 'Flower Care 01'
```

From HassOS commandline I can enter the bluetoothctl and do a "scan on" and see the device show up after a short wait. Can't do anything else beyond that from what I can see.

FYI, I have several other BLE devices (LYWSD03MMC temp sensors) that I have been using for months and they all work perfectly.

**Comment by Ernst79** [comment]
It can have several causes
- It looks like this is a rebrand of the Xiaomi plant sensor. Normally, it starts sending data after the sensor is added to the Xiaomi Mi Home app. I'm not sure if the Northfifteen app is triggering the sensor to send data is the Xiaomi MiBeacon format. E.g. Qingping (another chinese manufacturer) also has its own BLE advertisement format. So, the first thing you could try is adding it to the Xiaomi MiHome app. 
- Updating the firmware might help as well. It looks like it is using the Xiaomi firmware, which is also 3.2.2

**Comment by Krojack** [comment]
First thing I did was add it to the MiHome app. I still use the hacked one needed to get the encryption keys for LYWSD03MMC devices. It adds to the app but then errors out when trying to view the data. I then installed the "Flower Care" app that the device said to use. That's what I used to finally update the firmware.

So I went ahead and flashed the updated firmware from v2.7.0 up to v3.2.2. It's now working in HA. I saw an earlier update to this plugin saying it was fixed to work with v3.2.1 but wasn't sure if 3.2.2 would break. I'm always cautious about updating firmware in these situations.
